### ISSUE TITLE
Use youtube-dl to download audio mp3 file of You Tube live stream video in Bash file

### ISSUE THREAD

**Issue by LRose7** [matthewcarlino/tube-pod#1]

**Comment by matthewcarlino** [comment]
Hi @LRose7, I got something started with the node based [ytdl-core](https://github.com/fent/node-ytdl-core).  Just a basic getInfo call.  I've never used this before, but it did pull a massive blob of data on the recent sunday service on youtube.  There is an executable js file that should run in a shell.  I am using node 14.9.0.  Take a look and see where else you can go with it.

**Comment by LRose7** [comment]
Hey Matt, that is awesome! I will take a look and see what I can do 😊
-- 
 Securely sent with Tutanota. Get your own encrypted, ad-free mailbox: 
 https://tutanota.com


Aug 29, 2020, 13:25 by notifications@github.com:

**Comment by LRose7** [comment]
Good morning @matthewcarlino , I just wanted to update you on where I am with this project and in general.  God has really been working on me and growing me in the realm of communication as it is not my strong point.  I have been working on many projects and following along with some great Tutorials on You tube to build my skill sets in coding.  I have been learning so much and am growing.  Really been focusing on learning Vanilla JavaScript rather than frameworks like React and Angular so I can really understand the language.  Also been learning more of  Express.js and Node.js.  All that to say that I think this project is awesome and am very excited to be apart of it.  Every time I sit down to work on it I feel it is a little over my head at this point in time.   But I absolutely love coding and even though I have not wrote any lines of code yet for this project I have been working and studying behind the scenes to get better so I can contribute.  I am also learning what I love and I really enjoy working with the backend and building full stack applications.  So I just wanted to communicate all this to you and let you know that I do so very much want to help out and build things.  I feel like such a newbie and many times don't feel qualified but I am in awe of everything that I have learned thus far and how much God helps me.  I am hungry to learn and grow.  I hope all is well with you and your family.  May God continue to bless you all and keep you always.

**Comment by matthewcarlino** [comment]
Well @LRose7 I am glad to hear that you are still into it.  And I'm glad to hear you are learning a lot.  Since this is a command line tool, it's good that you are learning vanilla.  Our target runtime is a shell/cron job, so no need to involve the web.

I've written a bit more code... We can download an audio file, unfortunately, I cannot get it to play.  I get errors instead.  Maybe you could troubleshoot..

The [`ytdl`](https://github.com/fent/node-ytdl-core/blob/master/README.md#ytdlurl-options) function starts the download and returns a [readable stream](https://nodejs.org/api/stream.html#stream_class_stream_readable) which I then [pipe](https://nodejs.org/api/stream.html#stream_readable_pipe_destination_options) to a [writable stream](https://nodejs.org/api/stream.html#stream_class_stream_writable).

I suspect that I need some pass some options to the [`createWriteStream`](https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options) function or maybe there are more options to provide to the `ytdl` function.

Apart from the `ytdl` stuff, the rest is good fundamental NodeJS stuff.  All languages that I know of come with a "standard library" that has file system operations, readables and writables.

**Comment by LRose7** [comment]
@matthewcarlino Awesome I just did a git pull and am taking a look at it right now.  I am reading about the Cron daemon and Crontab Syntax. I just ran the code and the audio file downloaded but getting errors as well.  Will work on troubleshooting it.  I very much appreciate you talking through the code and saying what the target runtime is, that helps me very much.

**Comment by LRose7** [comment]
On my windows PC I ran the code after installing the dependencies and it downloaded the file with errors.  

On my Linux PC I ran the code and there was an error because there was no async function being used with await.  So I made an async function and ran the code again.  This time it downloaded but now the error is file is too big to be played.  

I haven't pushed anything to Github yet.  I am reading through the documentation and also found some similar projects on Github that are fairly new.

**Comment by LRose7** [comment]
@matthewcarlino I just made a push to GitHub.  I didn't change much but the code that is there downloads an mp3 file into my tube-pod project folder which I can play in the VLC media player on my computer.  The sound quality is great.

**Comment by LRose7** [comment]
@matthewcarlino the aysnc function to get video info pulls in a lot of metadata.  I noticed from the one url there is information on many different videos including some from C3 Church.  I also tried copying and pasting a video id from a c3 church sunday video to try and only get the meta data back for one specific video but it still returned a long list of all different videos.  I know that the code should come together and not be so separate but I was just experimenting with the getInfo and getBasicInfo to see what information would be shown.

**Comment by matthewcarlino** [comment]
Yeah.  I think all that data is basically a whole Youtube page in JSON format.  I believe most of those videos are "related videos".  The pertinent info is right at the top, I think.  It also breaks down the video into segments.

Regarding the mp3.  Quicktime seems to have trouble with it.  I'll keep troubleshooting that here.  VLC is super flexible, but unfortunately not what most people have.

While I do that, see if you can work on turning some of that JSON into an RSS feed item.  I'll make an "issue" for it.

**Comment by LRose7** [comment]
@matthewcarlino I found this video on youtube, https://www.youtube.com/watch?v=sL6SgDEHH00 (Working with RSS Feeds in Node.js)
I have been doing a lot of research on RSS Feeds.  Is this what you were envisioning?  I took his code from GitHub ( https://github.com/VLabStudio/Tutorials/blob/master/Working%20with%20RSS%20Feeds%20in%20Node.js/app.js ) but I am getting a 404 not found error so I am thinking I may have the the you tube channel id incorrect.  I am working on troubleshooting it now.  But just wanted to let you know where I am at.

**Comment by LRose7** [comment]
@matthewcarlino Just made a push to GItHub.  The error was because I did not have the proper you tube channel id for C3 Church LI.  Now when you run the code it creates a c3-church-li.json file with information for all the videos in the live stream playlist. Not quite sure if this is exactly what you wanted.
